File: Heartland-iOS-SDK/Classes/Terminals/Moby/Tem/SObjc.m
//
//  SObjc.m
//  Heartland-iOS-SDK
//
//

#import "SObjc.h"


@implementation SObjc

TemCommunicationManager * temCommunicationManager;


- (id) init
{
    self = [super init];
    temCommunicationManager = [[TemCommunicationManager alloc] init];
    return self;
}

+ (SObjc *)getInstance{
    static SObjc *sharedInstance = nil;
        static dispatch_once_t onceToken;
        dispatch_once(&onceToken, ^{
            sharedInstance = [[SObjc alloc] init];
        });
        return sharedInstance;
}


-(void)createSandbox:(NSString *)applicationId
          AndVersion:(NSString *)applicationVersion
               andIp:(NSString *)ip
             andport:(int)port
    andCallingMethod:(int)callingMethod
andAuthorizedActivities:(int)authorizedActivities
andOfflineInstallMode:(int)offlineInstallMode
   andContractNumber:(NSString *)contractNumber
              andSSL:(int)SSL
       andCertClient:(NSString *)certClient
        andKeyClient:(NSString *)keyClient
         andCertServ:(NSString *)CertServ
     andResultBlock : (void (^)(SandboxStatus))resultBlock{
    [temCommunicationManager initTemWithIp:ip andport:port andResultBlock:resultBlock];
}




- (void)createReaderContext:(NSString *)readerStateJson
        AndCompletionBlock : (void (^)(int))completionBlock{
    [temCommunicationManager createReaderContext:readerStateJson AndCompletionBlock:completionBlock];
}

- (void)updateReaderJsonState:(NSString *)readerStateJson AndCompletionBlock:(void (^)(int))completionBlock{
    [temCommunicationManager updateReaderStateJson:readerStateJson AndCompletionBlock:completionBlock];
}

- (void)performCall:(ReasonForCalling)reasonForCalling
 andCompletionBlock:(void (^)(int))completionBlock{
    if(reasonForCalling == ReasonForCallingMANUAL){
        [temCommunicationManager pollForUpdate:completionBlock];
    }else{
        [temCommunicationManager reportFirmwareStatus:completionBlock];
    }
}



-(void)isUpdateAvailable : (NSString *)deviceSerialNumber
          andResultBlock : (void (^)(bool))resultBlock{
    [temCommunicationManager isUpdateAvailable:deviceSerialNumber andResultBlock:resultBlock];
}


-(void)getUpdateFilePath : (NSString *)deviceSerialNumber andResultBlock : (void (^)(NSString *))resultBlock{
    return [temCommunicationManager getUpdateFilePath:deviceSerialNumber andResultBlock:resultBlock];
}

-(void)getRkiFileName : (NSString *)deviceSerialNumber andResultBlock : (void (^)(NSString *))resultBlock{
    return [temCommunicationManager getRkiFileName:deviceSerialNumber andResultBlock:resultBlock];
}

@end

File: Heartland-iOS-SDK/Classes/Terminals/Moby/Tem/SObjc.swift
//import Foundation
//
//class SObjc {
//    
//    private var temCommunicationManager: TemCommunicationManager
//    
//    init() {
//        temCommunicationManager = TemCommunicationManager()
//    }
//    
//    static let sharedInstance: SObjc = {
//        let instance = SObjc()
//        return instance
//    }()
//    
//    func createSandbox(applicationId: String,
//                       applicationVersion: String,
//                       ip: String,
//                       port: Int,
//                       callingMethod: Int,
//                       authorizedActivities: Int,
//                       offlineInstallMode: Int,
//                       contractNumber: String,
//                       SSL: Int,
//                       certClient: String,
//                       keyClient: String,
//                       certServ: String,
//                       resultBlock: @escaping (SandboxStatus) -> Void) {
//        temCommunicationManager.initTem(withIp: ip, andport: Int32(port), andResultBlock: resultBlock)
//    }
//    
//    func createReaderContext(readerStateJson: String,
//                             completionBlock: @escaping (Int) -> Void) {
//        temCommunicationManager.createReaderContext(readerStateJson) { value in
//            completionBlock(Int(value))
//        }
//    }
//    
//    func updateReaderJsonState(readerStateJson: String,
//                               completionBlock: @escaping (Int) -> Void) {
//        temCommunicationManager.updateReaderStateJson(readerStateJson) { value in
//            completionBlock(Int(value))
//        }
//    }
//    
//    func performCall(reasonForCalling: ReasonForCalling,
//                     completionBlock: @escaping (Int) -> Void) {
//        if reasonForCalling == .MANUAL {
//            temCommunicationManager.poll() { value in
//                completionBlock(Int(value))
//            }
//        } else {
//            temCommunicationManager.reportFirmwareStatus { value in
//                completionBlock(Int(value))
//            }
//        }
//    }
//    
//    func isUpdateAvailable(deviceSerialNumber: String,
//                           resultBlock: @escaping (Bool) -> Void) {
//        temCommunicationManager.isUpdateAvailable(deviceSerialNumber, andResultBlock: resultBlock)
//    }
//    
//    func getUpdateFilePath(deviceSerialNumber: String,
//                           resultBlock: @escaping (String) -> Void) {
//        temCommunicationManager.getUpdateFilePath(deviceSerialNumber, andResultBlock: resultBlock)
//    }
//    
//    func getRkiFileName(deviceSerialNumber: String,
//                        resultBlock: @escaping (String) -> Void) {
//        temCommunicationManager.getRkiFileName(deviceSerialNumber, andResultBlock: resultBlock)
//    }
//}
